cloudflare/resource_cloudflare_waf_package.go

```diff
@@ -0,0 +1,255 @@
+package cloudflare
+
+import (
+	"fmt"
+	"strings"
+
+	cloudflare "github.com/cloudflare/cloudflare-go"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceCloudflareWAFPackage() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceCloudflareWAFPackageCreate,
+		Read:   resourceCloudflareWAFPackageRead,
+		Update: resourceCloudflareWAFPackageUpdate,
+		Delete: resourceCloudflareWAFPackageDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: resourceCloudflareWAFPackageImport,
+		},
+
+		SchemaVersion: 0,
+		Schema: map[string]*schema.Schema{
+			"package_name": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+
+			"package_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+
+			"zone": {
+				Type:       schema.TypeString,
+				Optional:   true,
+				Computed:   true,
+				Deprecated: "`zone` is deprecated in favour of explicit `zone_id` and will be removed in the next major release",
+			},
+
+			"zone_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+
+			"sensitivity": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "high",
+			},
+
+			"action_mode": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "challenge",
```

Comment:
Similar to https://github.com/terraform-providers/terraform-provider-cloudflare/pull/475/files#r326930670, let's move towards better validation without a default.